src/common/snippets/src/lowered/linear_ir.cpp

```diff
@@ -178,13 +191,20 @@ void LinearIR::register_expression(const ExpressionPtr& expr, bool io_allowed, d
                     "LinearIR::insert can't be used to add Parameters or Results to IR");
     const auto& res = m_node2expression_map.insert({node, expr});
     OPENVINO_ASSERT(res.second, "Duplicate node is detected in linear IR: ", node);
+
+    expr->m_exec_num = exec_num;
+
     if (ov::is_type<ov::op::v0::Parameter>(node))
         m_parameter_expressions.push_back(expr);
     if (ov::is_type<ov::op::v0::Result>(node))
         m_result_expressions.push_back(expr);
-    if (ov::is_type<op::Buffer>(node))
-        m_buffer_expressions.push_back(expr);
-    expr->m_exec_num = exec_num;
+    if (const auto buffer_expr = ov::as_type_ptr<BufferExpression>(expr)) {
+        // just to align with execution order
+        auto it = m_buffer_expressions.cbegin();
+        while (it != m_buffer_expressions.cend() && expr->m_exec_num > (*it)->get_exec_num())
+            ++it;
```

Comment:
This is very inefficient. Why do we have to do it anyway? We didn't care too much before.
If we absolutely have to store the buffers sorted at all times, we need to implement a more optimal solution.

Reply:
Discussed offline: only SetBufferRegGroup requires ordered buffers -> now buffer container stores unsorted expressions but we locally sort them only in the needed pass. Thanks!